src/reanimated2/commonTypes.ts

```diff
@@ -1,47 +1,16 @@
-import type {
-  PerpectiveTransform,
-  RotateTransform,
-  RotateXTransform,
-  RotateYTransform,
-  RotateZTransform,
-  ScaleTransform,
-  ScaleXTransform,
-  ScaleYTransform,
-  TranslateXTransform,
-  TranslateYTransform,
-  SkewXTransform,
-  SkewYTransform,
-  MatrixTransform,
-  ViewStyle,
-  TextStyle,
-} from 'react-native';
-
-export type TransformProperty =
-  | PerpectiveTransform
-  | RotateTransform
-  | RotateXTransform
-  | RotateYTransform
-  | RotateZTransform
-  | ScaleTransform
-  | ScaleXTransform
-  | ScaleYTransform
-  | TranslateXTransform
-  | TranslateYTransform
-  | SkewXTransform
-  | SkewYTransform
-  | MatrixTransform;
+import type { ViewStyle, TextStyle } from 'react-native';
 
 export interface StyleProps extends ViewStyle, TextStyle {
   originX?: number;
   originY?: number;
   [key: string]: any;
 }
 
-export interface SharedValue<T> {
-  value: T;
-  addListener: (listenerID: number, listener: (value: T) => void) => void;
+export interface SharedValue<TValue> {
+  value: TValue;
```

Comment:
I think that `ValueT` instead of `TValue` would match our current style
```suggestion
export interface SharedValue<ValueT> {
  value: ValueT;
```